### PR TITLE
CFE-3273: Preparations for custom promise types (SeqSplit and parser refactoring)

### PR DESCRIPTION
**Pull request by olehermanse**
No change in behavior. Separate PR to split up test and review.

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom jenkins, please.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/5150/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/5150/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-5150/

File: libpromises/cf3parse_logic.h [hunk 1]
```diff
@@ -620,71 +620,70 @@ static inline void ParserAppendCurrentConstraint()
...
-        P.rval = RvalNew(NULL, RVAL_TYPE_NOPROMISEE);
-        strcpy(P.lval, "no lval");
-        RlistDestroy(P.currentRlist);
-        P.currentRlist = NULL;
-    }
-    else
-    {
-        RvalDestroy(P.rval);
-        P.rval = RvalNew(NULL, RVAL_TYPE_NOPROMISEE);
     }
```
Comment (olehermanse) [inline libpromises/cf3parse_logic.h]: This is the else which was moved to be an early return at the top.

**Comment by olehermanse** [comment]
> 1 for Pointer argument is dereferenced without checking for NULL

Introduced in libntech, not here.

**Pull request merged**